aten/src/ATen/native/cuda/UnaryOpsKernel.cu

```diff
@@ -47,6 +47,14 @@ void expm1_kernel_cuda(TensorIterator& iter) {
   });
 }
 
+void i0_kernel_cuda(TensorIterator& iter) {
+  AT_DISPATCH_FLOATING_TYPES_AND_HALF(iter.dtype(), "i0_cuda", [&]() {
```

Comment:
Could we add bfloat16 here and see it just work since bfloat16 uses float32 as its accumulate type?

https://github.com/pytorch/pytorch/blob/324c18fcad579b1afa63ae45528bf598ba8ec4ca/aten/src/ATen/AccumulateType.h#L25